sources/packages/backend/apps/workers/src/controllers/supporting-user/_test_/e2e/supporting-user.controller.createIdentifiableSupportingUsers.e2e-spec.ts

```diff
@@ -303,4 +351,47 @@ describe("SupportingUserController(e2e)-createIdentifiableSupportingUsers", () =
       resultType: MockedZeebeJobResult.Error,
     });
   });
+
+  /**
+   * Helper function to lookup notifications and perform assertions.
+   * @param savedApplication the saved application.
+   * @param parentFullName the full name of the parent.
+   * @param notificationMessageType the notification message type to check.
+   */
+  async function notificationLookupAndAssertion(
+    savedApplication: Application,
+    parentFullName: string,
+    notificationMessageType: NotificationMessageType,
+  ): Promise<void> {
+    const notification = await db.notification.findOne({
+      select: {
+        id: true,
+        dateSent: true,
+        messagePayload: true,
+        notificationMessage: { id: true, templateId: true },
+        user: { id: true, email: true, firstName: true, lastName: true },
+      },
+      relations: { notificationMessage: true, user: true },
+      where: {
+        notificationMessage: {
+          id: notificationMessageType,
+        },
+        user: {
+          id: savedApplication.student.user.id,
+        },
```

Comment:
If the data is controlled by the student user id, then I would say that update in before each(whole before each itself) is not necessary.